### PR TITLE
dials.refine scan_varying=auto for DIALS 2.0

### PR DESCRIPTION
**Pull request by dagewa**
This branch sets `scan_varying=auto` the default option for dials.refine. For scans, this performs `n_static_macrocycles` (default ==1) of static refinement followed by a round of scan-varying refinement. For stills, `n_static_macrocycles` also has effect, but there is no additional round of refinement to finish.

### CODE REVIEW

**Comment by graeme-winter** [comment]
@rjgildea proposed you as a reviewer - if you are happy please merge.

**Comment by graeme-winter** [comment]
Anything holding this one up?

File: command_line/refine.py [hunk 1]
```diff
@@ -116,411 +116,452 @@
...
 )
 
 # local overrides for refiner.phil_scope
-phil_overrides = parse(
+phil_overrides = libtbx.phil.parse(
     """
   refinement
   {
     verbosity = 2
+    parameterisation.scan_varying = Auto
```
Comment (rjgildea) [inline command_line/refine.py]: The help string should be updated to explain what the Auto option does.

**Comment by graeme-winter** [comment]
@rjgildea xia2 could cope with this by simply keeping the two steps and being explicit re: static / scan varying - so not hard. Rest of it seems fairly easy to resolve.

**Comment by graeme-winter** [comment]
Just ran xia2 against this branch and got

```
Command line: xia2 /Users/graeme/data/i04_bag_training/
-------------------- Spotfinding SWEEP1 --------------------
Error: type object 'reflection_table' has no attribute 'from_file'
Please send the contents of xia2.txt, xia2.error and xia2-debug.txt to:
xia2.support@gmail.com
```

so possibly needs a rebase? Will test out-of-xia2 land

**Comment by graeme-winter** [comment]
Re: xia2 / explicit 

```
dials.refine  '/Users/graeme/Projects/dials-pull-751/DEFAULT/NATIVE/SWEEP1/index/8_experiments_reindexed.json' '/Users/graeme/Projects/dials-pull-751/DEFAULT/NATIVE/SWEEP1/index/9_reflections_reindexed.pickle' 'scan_varying=False' 'close_to_spindle_cutoff=0.020000' 'outlier.algorithm=auto' 'output.experiments=10_refined_experiments.json' 'output.reflections=10_refined.pickle' 'reflections_per_degree=100' 'unit_cell.smoother.interval_width_degrees=36' 'orientation.smoother.interval_width_degrees=36'
```

already the case => should be harmless, but adapting xia2 to use this feature could be useful.

File: command_line/refine.py (hunk 1 above)
Comment (graeme-winter) [reply command_line/refine.py]: Resolved in https://github.com/dials/dials/pull/751/commits/ddcd0dd574ddb6b73868fba88145140805133d27

**Comment by graeme-winter** [comment]
Rebased against master to allow testing - passed just fine inside xia2, comparing the results now to ensure they are identical. 

FWIW will also require changes to DUI maybes? Though this is likely to make it much easier to use as the default behaviour in the work flow would now be _correct_

**Comment by graeme-winter** [comment]
Happy that this is a NO-OP for xia2:

```
Grey-Area dials-pull-751 :( $ diff xia2.txt master/xia2.txt 
6,9c6,9
< CCP4_SCR => /var/folders/hs/gvssfcd910s0jczjhjczj6h80000gn/T/tmpy7SLB6
< Starting directory: /Users/graeme/Projects/dials-pull-751
< Working directory: /Users/graeme/Projects/dials-pull-751
< Free space:        20.32 GB
---
> CCP4_SCR => /var/folders/hs/gvssfcd910s0jczjhjczj6h80000gn/T/tmpFMGS_h
> Starting directory: /Users/graeme/Projects/dials-pull-751/master
> Working directory: /Users/graeme/Projects/dials-pull-751/master
> Free space:        18.72 GB
13c13
< DIALS 2.dev.409-g968714149
---
> DIALS 2.dev.399-g8718ebd81
88c88
< MTZ file: /Users/graeme/Projects/dials-pull-751/DEFAULT/NATIVE/SWEEP1/integrate/dials_integrated.mtz
---
> MTZ file: /Users/graeme/Projects/dials-pull-751/master/DEFAULT/NATIVE/SWEEP1/integrate/dials_integrated.mtz
114c114
< Scaled reflections: /Users/graeme/Projects/dials-pull-751/DataFiles/AUTOMATIC_DEFAULT_free.mtz
---
> Scaled reflections: /Users/graeme/Projects/dials-pull-751/master/DataFiles/AUTOMATIC_DEFAULT_free.mtz
116,117c116,117
< Scaled reflections (NATIVE): /Users/graeme/Projects/dials-pull-751/DataFiles/AUTOMATIC_DEFAULT_scaled_unmerged.mtz
< Processing took 00h 08m 30s
---
> Scaled reflections (NATIVE): /Users/graeme/Projects/dials-pull-751/master/DataFiles/AUTOMATIC_DEFAULT_scaled_unmerged.mtz
> Processing took 00h 08m 26s

```

=> 👍

**Comment by graeme-winter** [comment]
Outstanding flake8 issues:

```Running flake8 on changed files:
algorithms/refinement/parameterisation/configure.py:639:39: F821 undefined name 'beam'
algorithms/refinement/parameterisation/configure.py:656:31: F821 undefined name 'beam'
```

these apply equally to master => ignoring for the moment; deal with in #809

**Pull request merged**